Create seeds

KoubaFromPucks opened the pull request:
Seeds are done. If you think it is not much data i can add some + tell me the case that is not implemented here (for example if there was no question of type multichoice).
Cases:
- 2 users (first has no picture, second user has pictureUrl)
- 2 Quizzes (first has 3 questions, second has 4 questions)
- There is 7 questions of all types. Some has just text, some has just picture and some has both.
- There are 3 completed quizzes. Somwhere is skipped question, some answers are ok, some answers are wrong.

KoubaFromPucks committed:
Add seed data
Seed data were added. There are 2 users. User 1 has completed quiz 1 with full score. User 2 has 0 points from quiz 1, where he skipped 1 question, and others are filled badly.
There 2 Quizzes. First contains just single/multi choice questions, some has picture url, some not.
Second Quiz has order/single/multi choice question. This quiz was filled just by second user.

Seeds are filling just Guids, not a navigation properties, as it is in CB.

brtkkndrl commented on src/Quizzer.API.DAL/Seeds/Storage.cs:
```diff
@@ -0,0 +1,559 @@
+            Questions.Add(new QuestionEntity {
+                Id = QuestionGuids[2],
+                Type = Enums.AnswerFormat.SingleChoice,
+                Text = "Is Austraia part of British commonwealth?",
```
typo

KoubaFromPucks replied:
fixed

brtkkndrl commented on src/Quizzer.API.DAL/Seeds/Storage.cs:
```diff
@@ -0,0 +1,559 @@
+                QuestionId = QuestionGuids[5]
+            });
+
+            UserAnswers.Add(new UserAnswerEntity {
```
Selected Answers don't need to be seeded here?

KoubaFromPucks replied:
No, it does not.

vecerap commented on src/Quizzer.API.DAL/Seeds/Storage.cs:
```diff
@@ -0,0 +1,559 @@
+        private void SeedQuizzes() {
+            Quizzes.Add(new QuizEntity {
+                Id = QuizGuids[0],
+                StartTime = DateTime.Parse("15.01.2023 00:00", System.Globalization.CultureInfo.CreateSpecificCulture("fr-FR")),
+                EndTime = DateTime.Parse("15.01.2024 19:34", System.Globalization.CultureInfo.CreateSpecificCulture("fr-FR")),
```
Why there is french culture?

KoubaFromPucks replied:
I like this format and i wanted to be sure, that those datetimes will be parsed correctly.

vecerap replied:
cs-cz does not have this format? I don't mind it that at all, but it surprised me choosing this culture.

vecerap reviewed (commented):
The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?

BRTEK007 commented:
> The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?

I agree.

vecerap commented:
> > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> 
> I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?

@KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list

vecerap commented:
> > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > 
> > 
> > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> 
> @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list

In code this could be `public Guid QuestionA => questionIds[2];`

KoubaFromPucks commented:
> > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > 
> > > 
> > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > 
> > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> 
> In code this could be `public Guid QuestionA => questionIds[2];`

Should this be made for every id? Because then the id lists are not really needed. Right?

KoubaFromPucks commented:
> > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > 
> > > > 
> > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > 
> > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > 
> > In code this could be `public Guid QuestionA => questionIds[2];`
> 
> Should this be made for every id? Because then the id lists are not really needed. Right?

And I guess that answer maybe really don't need this approach.

vecerap commented:
> > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > 
> > > > 
> > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > 
> > > 
> > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > 
> > 
> > In code this could be `public Guid QuestionA => questionIds[2];`
> 
> Should this be made for every id? Because then the id lists are not really needed. Right?

@KoubaFromPucks Well, I will leave it up to you, what stays is with names, the tests verification will be easier.

vecerap commented:
> > > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > > 
> > > > > 
> > > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > > 
> > > > 
> > > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > > 
> > > 
> > > In code this could be `public Guid QuestionA => questionIds[2];`
> > 
> > 
> > Should this be made for every id? Because then the id lists are not really needed. Right?
> 
> And I guess that answer maybe really don't need this approach.

I am not forcing to do it on every id, make it where it makes sense later during the tests creation.

KoubaFromPucks committed:
Add names to some ids from Id sets
In places where it is good to know, which id exactly is in some spot,
there are lambda functions that gives name to an id "unkown" id.
For example except of UserIds[0] there is a function named
UserKurtId, that returns id of user named kurt.

vecerap reviewed (approved):
Squash commit b97ae56359ae65cd5134942bdeab44262ab603f4 with 91d9834768a986a167acae8f6b9442bcc87c3e10. And keep message from 91d9834768a986a167acae8f6b9442bcc87c3e10

KoubaFromPucks commented:
> Squash commit [b97ae56](https://github.com/vecerap/iw5-2023/commit/b97ae56359ae65cd5134942bdeab44262ab603f4) with [91d9834](https://github.com/vecerap/iw5-2023/commit/91d9834768a986a167acae8f6b9442bcc87c3e10). And keep message from [91d9834](https://github.com/vecerap/iw5-2023/commit/91d9834768a986a167acae8f6b9442bcc87c3e10)

done
